march_rqt_gait_selection/src/march_rqt_gait_selection/gait_selection_view.py

```diff
@@ -0,0 +1,259 @@
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import \
+    QComboBox, QDialog, QGridLayout, QLabel, QPlainTextEdit, QPushButton, QScrollArea, QSizePolicy, QWidget
+from python_qt_binding import loadUi
+import rospy
+
+AMOUNT_OF_AVAILABLE_SUBGAITS = 5
+
+
+class GaitSelectionView(QWidget):
+    def __init__(self, ui_file, controller):
+        """Base class to load and use the gait_selection.ui.
+
+        :param ui_file:
+            Path to the .ui file to load with the widget
+        :param controller:
+            A refrence to the controller corresponding to the rqt gait selection functionalities
+        """
+        super(GaitSelectionView, self).__init__(flags=Qt.Window)
+
+        self._controller = controller
+
+        # Extend the widget with all attributes and children from UI file
+        loadUi(ui_file, self)
+
+        # Coding flags
+        self._colors = {'info': '#000000', 'success': '#009100', 'error': '#FF0000', 'warning': '#b27300'}
+        self._is_refresh_active = True
+        self._is_update_active = True
+
+        # Search for children in GUI
+        self._gait_content = self.Gaits.findChild(QScrollArea, 'GaitSelectionInterface')
+        self._gait_menu = self._gait_content.findChild(QComboBox, 'GaitMenu')
+        self._logger = self._gait_content.findChild(QPlainTextEdit, 'Log')
+
+        self._subgait_labels = []
+        self._subgait_menus = []
+
+        for index in range(AMOUNT_OF_AVAILABLE_SUBGAITS):
+            self._subgait_labels.append(self._gait_content.findChild(QLabel, 'SubgaitLabel_{nr}'.format(nr=index)))
+            self._subgait_menus.append(self._gait_content.findChild(QComboBox, 'SubgaitMenu_{nr}'.format(nr=index)))
+
+        # bind functions to callbacks of buttons and menus
+        self.findChild(QPushButton, 'Refresh').pressed.connect(
+            lambda: self._refresh())
+        self.findChild(QPushButton, 'Apply').pressed.connect(
+            lambda: [self._apply(), self._refresh()])
+        self.findChild(QPushButton, 'SaveDefault').pressed.connect(
+            lambda: [self._apply(), self._save_default(), self._refresh()])
+
+        self.findChild(QPushButton, 'ClearLogger').clicked.connect(lambda: self._logger.clear())
+        self.findChild(QPushButton, 'SeeAllVersions').clicked.connect(lambda: self._show_version_map_pop_up())
```

Comment:
Any particular reason you use `pressed` on the first 3 buttons and `clicked` on the last 2? Better to use one. I think we use `clicked` in other projects.

Reply:
yes, pressed kind of freezes the push button for as long as the function needs, this way the button cannot be clicked again when active. Not the most pretty solution but works kind of nice...

Reply:
Ah did not expect that :upside_down_face: